### DISCUSSION TITLE
New user, stuck on getting reclone remote to work.

### DISCUSSION THREAD

**Discussion by ehrie** [ep1cman/unifi-protect-backup#182] [Q&A]
Hello, new to this so if this is an easy question a link to a resource where I can figure it out would be just as helpful as a direct answer.

I am trying to set UPB up on TrueNAS Scale. I'm stuck when launching UPB that it throws this error:

ValueError: rclone does not have a remote called `unifi`

I created a remote using rclone config named that setting to it localdisk as I wish to back up to my TrueNAS machine, used the set of trace commands provided in the Debugging section and it in the listremotes it shows a remote named unifi. I am sure this is user error, but not sure where I have gone wrong or where to begin looking. Thanks.

**Comment by ehrie** [comment]
I believe I know what the problem is, but I am not experienced enough at this to know how to fix it properly. I am installing the app through the Discover Apps button in TrueNAS and I think it's setting the path to my rclone config file incorrectly. I see this in the UPB log:

cp: cannot stat '/root/.config/rclone/rclone.conf': No such file or directory

My rclone config file's path is /home/truenas_admin/.config/rclone/rclone.conf

When setting up the app from inside truenas scale I do not see a place to set the path for the rclone.conf file. How does a noob solve this?

**Comment by ehrie** [comment]
More work on this, but still can't get to the source. I moved my rclone config to the path UPB was expecting. Ran the debug command and listremotes returned the remote I created so I thought was good to go. Nope, still getting the exact same behavior. Same errors.

**Comment by ep1cman** [comment]
Could you post the full logs of the errors? And the output of your debug command?

**Reply by ehrie** [reply]
docker run \      
    --rm \
    -v /root/.config/rclone/rclone.conf:/config/rclone/rclone.conf \
    -e RCLONE_CONFIG='/config/rclone/rclone.conf' \
    --entrypoint rclone \
    ghcr.io/ep1cman/unifi-protect-backup \
    listremotes
unifi:

There's the debug command and what it spits out. Attached is the full log. Thanks for looking at this. I am sure there is something basic I did wrong, probably something with permissions.

[Error.txt](https://github.com/user-attachments/files/18267465/Error.txt)

**Reply by ep1cman** [reply]
Very odd, have you checked the file permissions?

**Reply by ehrie** [reply]
I set UPB as an admin in the TrueNAS GUI and that didn't change anything. Or I did it wrong, I don't know. I see in your docker install instructions there an argument for setting the path to rclone. This isn't present in the install method using the Discover Apps button, nor is in the setting screen inside the Scale GUI. Is there a way to manually change where the App is looking for rclone to be? When googling this I keep coming upon people having access issues because "root access to shares" is no longer supposed to Scale. I don't know if I am having the same issue the other people are having.

**Reply by ep1cman** [reply]
I'm afraid I don't maintain any of the trueNAS "apps" for UPB. Maybe you can find out who does and ask them for advice? 

If I had to guess what's going on: when you run the debug command you are running it as root and thus have permissions to access the file but when it is started as a service it's run as some other user. I would check the file permissions and make sure all users can access it.

There should be no need to change where UPB looks for the config file within the container. The way it's supposed to work is that you map this path within the container to some other path on the host (it does not have to be the same path as it is inside). Another thing you could try is mounting the whole directory not just the config file.

**Reply by ehrie** [reply]
Thanks for trying, I will post here for future users if I figure this out.

**Comment by ehrie** [comment]
It took a while, but I finally got this working by ditching TrueNAS Scale's built in version of this container and manually deploying it with Dockge using the instructions on this github. I can't find out who maintaining that for TrueNAS(they keep putting out small patches to it so someone is), but if anybody wanders here and has the same issues I was having, if you are trying to use TrueNAS Apps repository, ditch it and go with the way published here.
